src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -61,25 +64,32 @@ protected internal override void AddCallbackForPinValueChangedEvent(int pinNumbe
 
         private LibGpiodDriverEventHandler PopulateEventHandler(int pinNumber)
         {
-            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            lock (_pinNumberLock)
```

Comment:
Adding all these locks will hit performance, should we instead just make _pinNumberToSafeLineHandle a ConcurrentDictionary?

Reply:
I first considered that, but ConcurrentDictionary has a quite different API than Dictionary, so the two don't just work interchangeably. Also of course, a Concurrent<Something> doesn't help when we do `if (!HasKey(...) { Add(xxx) }` sequences. I'll take another look, though.

Reply:
Ok, I'm now using a combination of ConcurrentDictionary and a lock. IMHO timing in `OpenPin` and `ClosePin` methods is not critical. `Read` and `Write` don't need a lock now.